components/feature/session/src/main/java/mozilla/components/feature/session/PictureInPictureFeature.kt

```diff
@@ -39,7 +39,14 @@ class PictureInPictureFeature(
             sessionManager.runWithSessionIdOrSelected(customTabSessionId) { session ->
                 session.fullScreenMode
             }
-        return fullScreenMode && enterPipModeCompat()
+        return fullScreenMode && try {
+            enterPipModeCompat()
+        } catch (e: IllegalStateException) {
+            // On certain Samsung devices, if accessibility mode is enabled, this will throw an
+            // IllegalStateException even if we check for the system feature beforehand. So let's just
+            // catch it, and not enter PiP. See https://stackoverflow.com/q/55288858
+            false
```

Comment:
Nice find! Might be worth logging the error though with the crash reporter so we know how often this happens. `IllegalStateException` is quite broad of an exception so we may catch other unknown exceptions here too.